### PR TITLE
implement use client protocol in destination rule

### PR DESCRIPTION
**Pull request by ramaraochavali**
Implements the api https://github.com/istio/api/pull/1627

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by brian-avery — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -573,6 +573,13 @@ func setH2Options(cluster *cluster.Cluster) {
 
 func applyTrafficPolicy(opts buildClusterOpts) {
 	connectionPool, outlierDetection, loadBalancer, tls := selectTrafficPolicyComponents(opts.policy)
+	if opts.direction == model.TrafficDirectionOutbound && connectionPool != nil && connectionPool.Http != nil && connectionPool.Http.UseClientProtocol {
+		setH2Options(opts.cluster)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:577]: not sure why it is here? IIUC `applyH2Upgrade` has already done this

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:577]: applyH2Upgrade is not applicable when useClientProtocol is set.

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
Do we have an integration test?

**Comment by ramaraochavali** [comment]
> Do we have an integration test?

We do not have. I can add one if it is useful?

**Comment by hzxuzhonghu** [comment]
It's better to have one

**Comment by ramaraochavali** [comment]
Will add in a follow-up

**Pull request merged**